syntax = "proto3";
package server_events;

import "Protobuf/minigame_responses.proto";

message ServerEventWrapper {
  oneof type {
    RoomInfo room_info = 1;
    RoomsList rooms_list = 2;
    minigame_responses.MinigameResponse game_response = 4;
  }
}

message RoomUser {
  enum Status {
    kNone = 0;
    kNotReady = 1;
    kReady = 2;
  }
  string nickname = 1;
  Status is_ready = 2;
}

message RoomInfo {
  repeated RoomUser users = 1;
  uint64 room_id = 2;
  uint64 chief_id = 3;
}

message RoomsList {
  repeated uint64 ids = 1;
}

message MinigamesList {
  repeated uint64 ids = 1;
}

message GameStatus {
  enum Status {
    kStarted = 0;
    kGoing = 1;
    kEnded = 2;
  }
  uint64 progress = 1;
  MinigamesList minigames_list = 2;
  Status status = 3;
}
